src/components/FirmwareUpdateBanner.tsx

```diff
@@ -1,86 +1,106 @@
 import {
   lastSeenDeviceSelector,
   hasCompletedOnboardingSelector,
+  lastConnectedDeviceSelector,
 } from "../reducers/settings";
 import { hasConnectedDeviceSelector } from "../reducers/appstate";
-import { BaseButton as Button } from "./Button";
+import Button from "./Button";
+import useEnv from "@ledgerhq/live-common/lib/hooks/useEnv";
+import { useFeature } from "@ledgerhq/live-common/lib/featureFlags";
+import useLatestFirmware from "../hooks/useLatestFirmware";
+import { gte as isVersionGreaterOrEqual } from "semver";
+import { DeviceModelId } from "@ledgerhq/devices";
+
+const devicesWithFwVersionConstraints: (DeviceModelId | undefined)[] = [
+  DeviceModelId.nanoS,
+  DeviceModelId.nanoX,
+];
 
 const FirmwareUpdateBanner = () => {
   const lastSeenDevice: DeviceModelInfo | null = useSelector(
     lastSeenDeviceSelector,
   );
+  const lastConnectedDevice = useSelector(lastConnectedDeviceSelector);
   const hasConnectedDevice = useSelector(hasConnectedDeviceSelector);
   const hasCompletedOnboarding: boolean = useSelector(
     hasCompletedOnboardingSelector,
   );
+
+  const navigation = useNavigation();
   const [showDrawer, setShowDrawer] = useState<boolean>(false);
-  const [showBanner, setShowBanner] = useState<boolean>(false);
-  const [version, setVersion] = useState<string>("");
 
   const { colors } = useTheme();
   const { t } = useTranslation();
-  const useTouchable = useContext(ButtonUseTouchable);
-
-  useEffect(() => {
-    async function getLatestFirmwareForDevice() {
-      const fw:
-        | FirmwareUpdateContext
-        | null
-        | undefined = await manager.getLatestFirmwareForDevice(
-        lastSeenDevice?.deviceInfo,
-      );
 
-      setShowBanner(Boolean(fw));
-      setVersion(fw?.final?.name ?? "");
-    }
+  const onExperimentalFirmwareUpdate = useCallback(() => {
+    navigation.navigate(NavigatorName.Manager, {
+      screen: ScreenName.Manager,
+      params: { firmwareUpdate: true },
+    });
+    setShowDrawer(false);
+  }, [navigation]);
 
-    getLatestFirmwareForDevice();
-  }, [lastSeenDevice, setShowBanner, setVersion]);
+  const latestFirmware = useLatestFirmware(lastSeenDevice?.deviceInfo);
+  const showBanner = Boolean(latestFirmware);
+  const version = latestFirmware?.final?.name ?? "";
 
   const onPress = () => {
     setShowDrawer(true);
   };
   const onCloseDrawer = () => {
     setShowDrawer(false);
   };
-  const onDismissBanner = () => {
-    setShowBanner(false);
-  };
 
-  return showBanner && hasConnectedDevice && hasCompletedOnboarding ? (
+  const usbFwUpdateExperimental = useEnv("USB_FW_UPDATE");
+  const usbFwUpdateFeatureFlag = useFeature("llmUsbFirmwareUpdate");
+  const isUsbFwUpdateFeatureActivated =
+    usbFwUpdateExperimental || usbFwUpdateFeatureFlag?.enabled;
+  const isUsbFwVersionUpdateSupported =
+    latestFirmware?.final?.version &&
+    (isVersionGreaterOrEqual(latestFirmware.final.version, "2.0.0") ||
```

Comment:
Aren't we checking that the _target_ version is greater than 2.0.0 here and not that the current one is greater than that list we had from Benito? https://ledger.slack.com/archives/C038GTZBER5/p1650445219900559?thread_ts=1650444510.111489&cid=C038GTZBER5

We need to check that the device is running already a version above that, not that the target version for the update is greater than it, if that makes sense.

Reply:
Done